lizard_auth_server/forms.py

```diff
@@ -42,6 +85,57 @@ def clean(self):
         return new_data
 
 
+class JWTDecryptForm(forms.Form):
+    """Form to decrypt and verify JWT requests."""
+    key = forms.CharField(max_length=1024)
+    message = JWTField(max_length=8192,
+                       allowed_keys=('key', 'domain', 'force_sso_login'))
+
+    def __init__(self, *args, **kwargs):
```

Comment:
@byrman, kun jij je blik laten gaan over deze `__init__()` plus de `clean()` in combinatie met het JWTField hierboven?

Probleem is uiteindelijk dat je twee velden hebt die met elkaar te maken hebben. In een form met password1 en password2 kan je pas in de `clean()` van het form de twee waardes met elkaar vergelijken.

In dit geval komt er een form binnen met een veld `key` en een `message`. De key is nodig om een site object op te halen waar een sso_key aan zit. Daarmee moet de inhoud van de message geverifieerd worden.

Alternatief "Reinout": de jwt message in JWTField uitpakken en de resultaten als dict teruggeven, maar nog geen signature controleren, want die heb je nog niet. In de `clean()` van het form moet je dan een extra `self.fields['message'].verify_signature(secret_key_die_ik_nu_heb)` doen. En die moet je niet vergeten, want anders is het niet veilig.

Alternatief Jackie zie je hier in het pull request. Je moet het 'key' veld in de init van het form al uitlezen (voordat de eigenlijke validatiestappen zijn geweest), en daarmee de secret key ophalen en die specifiek in de JWTField instantie stoppen. Als het daar mis mee gaat moet je de validatie errors bewaren, want die kan je niet in de `__init__()` al raisen. Vervolgens moet je die in de `validate()` weer teruggeven.

Jackie gruwt van mijn oplossing, want je kan de `verify_signature()` aanroep vergeten en dat is een enorm veiligheidslek.

Ik gruw van Jackie's oplossing, want ik vind het er lelijk uitzien en de dingen die in de init gedaan moeten worden staan me niet aan.

Wat vind jij? :-)

Op de manier die Jackie nu heeft geïmplementeert